Basic logos storage roadmap (#296)
logos-co/roadmap: content/storage/roadmap/privacy-preserving-filesharing.md

# Privacy-Preserving File Sharing

**Estimated date of completion.** Oct 2026

Privacy-preserving file sharing means that:

> * neither publisher or downloader of a file can be identified as so by third parties when querying and/or retrieving data;

Reviewer: ```suggestion
* neither publisher or downloader of a file can be identified by third parties when querying and/or retrieving data;
```

> * cache nodes cannot be held accountable for the content they cache (plausible deniability).

Reviewer: ```suggestion
* cache nodes should be able to convincingly deny knowledge of or responsibility for content they cache (plausible deniability).
```

* lessons from Freenet Routing.[^sandberg_05][^freenet_25]

We then need to build an understanding of the characteristics and tradeoffs for each approach. Tribler for instance explicitly states in their website that their routing is not meant to protect users from government-grade adversaries, so we must develop a good understanding of what that means. There is also abundant literature describing how Tor might be vulnerable to powerful adversaries.[^johnson_13][^gegenhuber_23]

This understanding, together with what compromises we are (and are not) willing to make should ultimately dictate how we balance performance, privacy, and a timely release schedule in providing this.

## Risks

| Risk                   | (Accept, Own, Mitigation)                                                                                        |
|------------------------|------------------------------------------------------------------------------------------------------------------|
| We take an approach of building filesharing first and then enhancing it with privacy, and figure out that we need an different protocol altogether. | Try to identify this as early as possible and fail fast. |

> | Anon coms team builds something that does not work for us. | Close collaboration with anon coms, with the embedding of a logos storage researcher within their effort. Employ strategic and small multidisciplinary team including both cryptography and networking researchers and engineers. |

Reviewer: ```suggestion
| Anon comms team builds something that does not work for us. | Close collaboration with anon comms, with the embedding of a logos storage researcher within their effort. Employ strategic and small multidisciplinary team including both cryptography and networking researchers and engineers. |
```
